### PR TITLE
[Auditbeat] Socket: Add network.transport and network.community_id

### PR DESCRIPTION
**Pull request by cwurm**
Adds `network.transport` (always `tcp` at the moment) and `network.community_id` to the `socket` dataset.

I tested that the community ID here is identical with the one generated by Packetbeat and that it can be used to match Packetbeat data (e.g. `flow`, `http`) with Auditbeat `socket` data to get the process and user information.

### CODE REVIEW

File: x-pack/auditbeat/module/system/socket/socket_test.go [hunk 1]
```diff
@@ -107,7 +108,13 @@ func TestOutbound(t *testing.T) {
 	checkFieldValue(t, event.RootFields, "process.name", "socket.test")
 	checkFieldValue(t, event.RootFields, "user.id", os.Geteuid())
 	checkFieldValue(t, event.RootFields, "network.direction", sock.Outbound.String())
+	checkFieldValue(t, event.RootFields, "network.transport", "tcp")
 	checkFieldValue(t, event.RootFields, "destination.port", 80)
+
+	communityID, err := event.RootFields.GetValue("network.community_id")
+	if assert.NoError(t, err) {
+		assert.NotEmpty(t, communityID)
```
Comment (andrewkroh) [inline x-pack/auditbeat/module/system/socket/socket_test.go]: I recommend checking the value here too just to make sure that if someone refactors this in the future that they don't jumble any of the fields.

Comment (cwurm) [reply x-pack/auditbeat/module/system/socket/socket_test.go]: The value depends on what `google.com` resolves to. I think we could calculate the community ID separately in the test and compare it to the one that the dataset generates if you think that makes sense?

Comment (andrewkroh) [reply x-pack/auditbeat/module/system/socket/socket_test.go]: Oh, I misread the diff. I thought this test was with the static endpoints created by `testSocket()`. Nevermind unless there is a different place to check it that is static.

Comment (cwurm) [reply x-pack/auditbeat/module/system/socket/socket_test.go]: Last time I touched this code I tried to write a test that creates two sockets, but it quickly grew complex to do that in one function (goroutines, done channels, etc.). So there is no place at the moment where we have a predictable pair of IPs.

Comment (andrewkroh) [reply x-pack/auditbeat/module/system/socket/socket_test.go]: I was thinking something like

```
s := testSocket()
assert.Equal(t, "1:abc", s.communityID())
```

Comment (cwurm) [reply x-pack/auditbeat/module/system/socket/socket_test.go]: Makes sense, I've added a test.

**Pull request merged**